frontend/src/components/Common/Chart/index.tsx

```diff
@@ -125,79 +120,73 @@ const Chart = memo(
           }),
         )
       );
-    }, [colorShuffle, config.colors, nshades]);
+    }, [config.colors, colorShuffle, nshades]);
 
-    const labels = React.useMemo(() => {
-      if (!transpose) {
-        return indices.map(index => header[index]);
-      }
-      return tableRows
-        .slice(chartRange[0], chartRange[1])
-        .map(row => row[config.category]);
-    }, [chartRange, config.category, header, indices, tableRows, transpose]);
+    const labels = useMemo(() => {
+      return transpose
```

Comment:
@Bhavyajain21 in some cases in this PR you are changing to ternary tetss and in others you are changing a ternary to an if statement, can you explain the rationale behind these changes?